lib/exchange_api/accounts.ex

```diff
@@ -52,8 +53,11 @@ defmodule ExchangeApi.Accounts do
 
   """
   def create_user(attrs \\ %{}) do
-    {:ok, token, _claims} = Guardian.encode_and_sign(user)
-    attrs = Map.put(user_params, "jwt", token)
+    uuid = Ecto.UUID.generate()
+
+    {:ok, token, _} = Guardian.encode_and_sign(uuid)
+    attrs = Map.put(attrs, "uuid", uuid)
+    attrs = Map.put(attrs, "jwt", token)
 
     %User{}
```

Comment:
Your `User.changeset` won't cast the uuid into the struct so you should initialize it with the id rather than passing it in your params.

```
%User{id: uuid}
|> User.changeset(attrs)
```